ktor-server/ktor-server-plugins/ktor-server-sse/common/test/io/ktor/server/sse/ServerSentEventsTest.kt

```diff
@@ -291,6 +298,31 @@ class ServerSentEventsTest {
         )
     }
 
+    @Test
+    fun testHeartbeat() = testApplication {
+        install(SSE)
+        routing {
+            sse {
+                heartbeat {
+                    duration = 100.milliseconds
+                    event = ServerSentEvent("heartbeat")
+                }
+
+                send("Hello")
+                delay(200.milliseconds)
+            }
+        }
+
+        val client = createSseClient()
+        val events = mutableListOf<ServerSentEvent>()
+        client.sse {
+            incoming.collect { events.add(it) }
+        }
+        assertTrue { events.size > 1 }
+        assertTrue { events.filter { it.data == "Hello" }.size == 1 }
+        assertTrue { events.any { it.data == "heartbeat" } }
+    }
```

Comment:
Maybe instead of using delays and asserts here, you could use some async processing and count a few of each to cover receiving multiple...

Something like:

```kotlin

    @Test
    fun testHeartbeat() = testApplication {
        install(SSE)
        routing {
            sse {
                heartbeat {
                    duration = 10.milliseconds
                    event = ServerSentEvent("heartbeat")
                }

                while(true) {
                    send("Hello")
                    delay(10.milliseconds)
                }
            }
        }

        val client = createSseClient()
        
        var hellos = 0
        var heartbeats = 0
        withTimeout(1_000) {
            client.sse {
                incoming.collect { event ->
                    when(event.data) {
                        "Hello" -> hellos++
                        "heartbeat" -> heartbeats++
                    }
                    if (hellos > 3 && heartbeats > 3)
                        cancel()
                }
            }
        }
    }
```

Reply:
Looks better, thanks!